test/jdk/java/util/concurrent/tck/SynchronousQueueTest.java

```diff
@@ -653,4 +657,64 @@ public void testNeverContainsNull() {
         assertFalse(q.remove(null));
     }
 
+    public void testFairDoesntLeak() throws InterruptedException {
+        assertDoesntLeak(new SynchronousQueue<>(true));
+    }
+
+    public void testUnfairDoesntLeak() throws InterruptedException {
+        assertDoesntLeak(new SynchronousQueue<>(false));
+    }
+
+    private void assertDoesntLeak(SynchronousQueue<Object> queue) throws InterruptedException {
```

Comment:
@DougLea @AlanBateman Decided to create a memory leak detection test for SynchronousQueue. Of course, it will likely take some time to tune the parameters of it, but if it completes successfully there was no leak of queue items tho...